control/velocity_controller/src/velocity_controller_core.cpp

```diff
@@ -219,60 +234,109 @@ void VelocityController::callbackTimerControl(const ros::TimerEvent & event)
+    set_double_param_if_changed("weak_brake_time", p.weak_brake_time);
+    set_double_param_if_changed("weak_brake_acc", p.weak_brake_acc);
+    set_double_param_if_changed("increasing_brake_time", p.increasing_brake_time);
+    set_double_param_if_changed("increasing_brake_gradient", p.increasing_brake_gradient);
+    set_double_param_if_changed("stop_brake_time", p.stop_brake_time);
+    set_double_param_if_changed("stop_brake_acc", p.stop_brake_acc);
+  }
 
   // acceleration limit
-  max_acc_ = config.max_acc;
-  min_acc_ = config.min_acc;
+  set_double_param_if_changed("max_acc", max_acc_);
+  set_double_param_if_changed("min_acc", min_acc_);
 
   // jerk limit
-  max_jerk_ = config.max_jerk;
-  min_jerk_ = config.min_jerk;
+  set_double_param_if_changed("max_jerk", max_jerk_);
+  set_double_param_if_changed("min_jerk", min_jerk_);
 
   // slope compensation
-  max_pitch_rad_ = config.max_pitch_rad;
-  min_pitch_rad_ = config.min_pitch_rad;
-  current_vel_threshold_pid_integrate_ = config.current_velocity_threshold_pid_integration;
+  set_double_param_if_changed("max_pitch_rad", max_pitch_rad_);
+  set_double_param_if_changed("min_pitch_rad", min_pitch_rad_);
+  set_double_param_if_changed(
+    "current_velocity_threshold_pid_integration", current_vel_threshold_pid_integrate_);
 
-  lpf_pitch_.init(config.lpf_pitch_gain);
+  {
+    double lpf_pitch_gain = get_parameter("lpf_pitch_gain").as_double();
+    set_double_param_if_changed("lpf_pitch_gain", lpf_pitch_gain);
+    lpf_pitch_.init(lpf_pitch_gain);
+  }
 
   // velocity feedback
-  pid_vel_.setGains(config.kp, config.ki, config.kd);
-  pid_vel_.setLimits(
-    config.max_out, config.min_out, config.max_p_effort, config.min_p_effort, config.max_i_effort,
-    config.min_i_effort, config.max_d_effort, config.min_d_effort);
-  lpf_vel_error_.init(config.lpf_velocity_error_gain);
+  // PID gain
+  {
+    double kp = get_parameter("pid_controller/kp").as_double();
+    double ki = get_parameter("pid_controller/ki").as_double();
+    double kd = get_parameter("pid_controller/kd").as_double();
+    set_double_param_if_changed("pid_controller/kp", kp);
+    set_double_param_if_changed("pid_controller/ki", ki);
+    set_double_param_if_changed("pid_controller/kd", kd);
+    pid_vel_.setGains(kp, ki, kd);
+  }
+
+  // PID limits
+  {
+    double max_pid = get_parameter("pid_controller/max_out").as_double();
+    double min_pid = get_parameter("pid_controller/min_out").as_double();
+    double max_p = get_parameter("pid_controller/max_p_effort").as_double();
+    double min_p = get_parameter("pid_controller/min_p_effort").as_double();
+    double max_i = get_parameter("pid_controller/max_i_effort").as_double();
+    double min_i = get_parameter("pid_controller/min_i_effort").as_double();
+    double max_d = get_parameter("pid_controller/max_d_effort").as_double();
+    double min_d = get_parameter("pid_controller/min_d_effort").as_double();
```

Comment:
That's a lot of code duplication. It would be much easier to put the two operations `get` and `set` into a function

Reply:
The setter function is already implemented in the original src code, and we need to modify them to combine with the new getter function defined here.

There should be a more effective way, e.g. having these parameters in the class member, and it should be done after porting. I'll write it as a TODO comment here.

Reply:
alright, go ahead for now @TakaHoribe 
I'm having the same issue right now and found a different solution. Will ping you when the PR is ready :)